h2o-core/src/main/java/water/TimeLine.java

```diff
@@ -184,7 +184,7 @@ public static InetAddress inet( long[] tl, int idx ) {
   // possible to the same point in time.
   static long[][] SNAPSHOT;
   static long TIME_LAST_SNAPSHOT = 1;
-  static private H2O CLOUD;      // Cloud instance being snapshotted
+  static private H2O CLOUD = H2O.CLOUD;      // Cloud instance being snapshotted
```

Comment:
Is this needed just in tests or also for regular execution?

Reply:
See this code
```
        for( int i=0; i<CLOUD._memary.length; i++ )
          if( CLOUD._memary[i]==H2O.SELF )
            SNAPSHOT[i] = a;
        TimeLine.class.notify();
```

the code works only because the initial CLOUD doesn't have any members, otherwise, there would be NPE because snapshot is not initialized. This means the code will not actually doing anything.

Both `call` and `tcp_call` methods don't seem to do anything in that case. I think this should be made more obvious by handling CLOUD != null in those methods and returning without action.

@jakubhava @honzasterba WDYT?

Reply:
@michalkurka It happens also in normal exeution. Agree ^ Will have a look